bin/nodejs-setup

```diff
@@ -0,0 +1,304 @@
+#!/bin/bash
```

Comment:
Am I right that this script came from somewhere else?  Is it possible to use it from its source rather than having a copy here?

Reply:
@nedbat This comes from the (sadly, all-too-prevalent) "curl and pipe to sudo/bash` idiom [1] :\

```sh
curl -sL https://deb.nodesource.com/setup_11.x | sudo -E bash -
```

It's possible, but I'd prefer to avoid propagating this pattern.
Though to be fair, the script itself contains some similar logic.

On the other hand, using a local file _might_ (?) improve the cache-ability of the layer when built by Docker

🤷‍♂ 

- [1] https://github.com/stvstnfrd/xblock-sdk/blob/nodejs/bin/nodejs-setup#L6-L13

Reply:
Copying the script in also has the downside of it potentially getting stale/breaking compatibility with the nodesource.com, seems like... What are the downsides of propagating the curl and pipe to sudo/bash line here?  It is the recommended installation method, after all: https://github.com/nodesource/distributions

One other thought: node 11 is out of LTS/isn't the current version... I think what we probably want is to use node 10 (node only seems to support even numbered versions in LTS for any length of time).  Node 12 is more of the bleeding edge, and FWIW edX is in the process of making sure all our JS code works in 10 still.

Reply:
@davidjoy Done. It now `curl`s down the node 10 install script and runs it, instead of a locally cached copy.

CC: @nedbat